include/deal.II/grid/tria_boundary.h

```diff
@@ -361,6 +367,14 @@ class StraightBoundary : public Boundary<dim,spacedim>
   Point<spacedim>
   project_to_surface (const typename Triangulation<dim,spacedim>::hex_iterator &hex,
                       const Point<spacedim> &candidate) const;
+
+  /**
+   * Same function as above for arbitrary surrounding points.
+   */
+  virtual
+  Point<spacedim>
+  project_to_manifold (const std::vector<Point<spacedim> > &surrounding_points,
```

Comment:
I would have suggested to actually make this the primary function, and document the others in terms of this one (rather than the other way around). But since we're going to get rid of this entire class at one time in the future, I don't think it's a big deal one way or the other.